packages/next/test/fixtures/00-server-build/index.test.js

```diff
@@ -1,8 +1,144 @@
+        console.log('checking', {
+          urlPath,
+          query,
+          initialRandom,
+          preRevalidateRandom,
+        });
+
+        if (preRevalidateRandom) {
+          // wait for change as cache may take a little to propagate
+          const initialUrl = `${ctx.deploymentUrl}${urlPath}${query || ''}`;
+          await checkForChange(initialUrl, preRevalidateRandom, async () => {
+            const res = await fetch(initialUrl);
+            const $ = cheerio.load(await res.text());
+            return JSON.parse($('#props').text()).random;
+          });
+        }
+
+        const res = await fetch(`${ctx.deploymentUrl}${urlPath}${query || ''}`);
+        expect(res.status).toBe(200);
+
+        const $ = await cheerio.load(await res.text());
+        const props = JSON.parse($('#props').text());
+
+        if (initialRandom) {
+          // for fallback paths the initial value is generated
+          // in the foreground and then a revalidation is kicked off
+          // in the background so the initial value will be replaced
+          if (initialRandom !== props.random && urlPath.includes('on-demand')) {
+            initialRandom = props.random;
+          } else {
+            expect(initialRandom).toBe(props.random);
+          }
+        } else {
+          initialRandom = props.random;
+        }
+        expect(isNaN(initialRandom)).toBe(false);
+
+        const dataRes = await fetch(
+          `${ctx.deploymentUrl}/_next/data/testing-build-id${urlPath}.json${
+            query || ''
+          }`
+        );
+        expect(dataRes.status).toBe(200);
+
+        const { pageProps: dataProps } = await dataRes.json();
+
+        if (initialRandomData) {
+          // for fallback paths the initial value is generated
+          // in the foreground and then a revalidation is kicked off
+          // in the background so the initial value will be replaced
+          if (
+            initialRandomData !== dataProps.random &&
+            urlPath.includes('on-demand-2')
+          ) {
+            initialRandomData = dataProps.random;
+          } else {
+            expect(initialRandomData).toBe(dataProps.random);
+          }
+        } else {
+          initialRandomData = dataProps.random;
+        }
+        expect(isNaN(initialRandomData)).toBe(false);
+      }
+    };
+
+    await checkPaths(pathsToCheck);
+
+    preRevalidateRandom = initialRandom;
+    preRevalidateRandomData = initialRandomData;
+
+    initialRandom = undefined;
+    initialRandomData = undefined;
+
+    const revalidateRes = await fetch(
+      `${ctx.deploymentUrl}/api/revalidate?urlPath=${pathsToCheck[0].urlPath}`
+    );
+    expect(revalidateRes.status).toBe(200);
+    expect((await revalidateRes.json()).revalidated).toBe(true);
+
+    await checkPaths(pathsToCheck);
+
+    expect(preRevalidateRandom).not.toBe(initialRandom);
+    expect(preRevalidateRandomData).not.toBe(initialRandomData);
```

Comment:
Since both these vars are undefined at this point, these two assertions are equivalent to the following, right?

```js
expect(preRevalidateRandom).toBeDefined()
expect(preRevalidateRandomData).toBeDefined()
```

Reply:
Yeah should be the same, updated in https://github.com/vercel/vercel/pull/8158/commits/a15615a51f554eeb93d401639030e61b6d8a57e9 for clarity